openlibrary/plugins/openlibrary/pd.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+
+import requests
+
+from infogami import config
+from openlibrary.core import cache
+from openlibrary.i18n import gettext as _
+from openlibrary.utils.dateutil import DAY_SECS
+
+
+@dataclass
+class PDOption:
+    """Represents an option in a print-disability qualifying organization select element"""
+
+    label: str
+    value: str
+
+
+def get_pd_options() -> list[PDOption]:
+    """Returns a list of all print-disability qualifying organization options"""
+    options = [
+        PDOption("BARD", "ia_nlsbardaccess_disabilityresources"),
+        PDOption("BookShare", "ia_bookshareaccess_disabilityresources"),
+        PDOption("ACE", "aceportalocul_disabilityresources"),
+        PDOption(_("I don't have one yet"), "unqualified"),
+    ]
+
+    pd_orgs = cached_pd_org_query()
+    options += [PDOption(org.get("title"), org.get("identifier")) for org in pd_orgs]
+
+    return options
+
+
+def get_pd_org(identifier: str) -> str | None:
+    """Returns the name of the organization associated with the given identifier.
+    Falls back to vtmas_disabilityresources if no match is found.
+    """
+    orgs = cached_pd_org_query()
+    vtmas = None
+    for org in orgs:
+        if org['identifier'] == identifier:
+            return org
+        if org['identifier'] == 'vtmas_disabilityresources':
+            vtmas = org
```

Comment:
```suggestion
    vtmas = {
        "identifier": "vtmas_disabilityresources",
        "title": "Vermont Mutual Aid Society",
    }
    for org in orgs:
        if org['identifier'] == identifier:
            return org

```
We could do something like this to ensure that a fallback organization is returned when the PD organization query fails.